packages/appeals-service-api/src/mappers/questionnaire-submission/has-mapper.js

```diff
@@ -0,0 +1,114 @@
+class HasQuestionnaireMapper {
+	mapToPINSDataModel(journeyResponse) {
+		return [
+			{
+				questionnaire: {
+					LPACode: journeyResponse.LPACode,
+					caseReference: journeyResponse.caseReference,
+					isAppealTypeAppropriate: this.#convertToBoolean(
+						journeyResponse.answers['correct-appeal-type']
```

Comment:
would it be possible to get these fieldName values directly from `questions.appealTypeAppropriate.fieldName` etc.?

Reply:
It would, and I think Bryn has raised this below and it is something I considered. My thinking is that we'll have to do something similar for S78 and we can try to scope out opportunities for refactoring at that point, and that would be one of them I think.

Reply:
I think there's at least three things here:
1: only sending through answers to questions that are conditionally valid for the journey based on the response. This should always be the same as we control what questions are valid to have been answered
2: lookup fieldnames that are hardcoded strings here
3: modifying the fieldnames/values/structure/metadata based on data model/requirements from back office

moving the journey to shared business rule can solve 1/2
this mapper can then still handle any requirements of 3

Happy for it to be a refactor job though

Reply:
Yeah I think it would have to be a refactor and we'd need to get some requirements from s78 as the journey there is more complex in comparison to HAS so right now I don't think we'd capture everything, you're still creating a coupling between the forms web app, the api and the pins data model it's just a case of identifying that risk and how to mitigate it. I'm imagining potential for the journey to become increasingly complex to serve both sub systems, but that's based purely on the fact of only having seen the HAS questionnaire and the WIP in progress for S78 and not knowing a great deal yet about other journeys and or journeys for other users, plus with the data model still in development there's a lot of unknowns for me that makes a refactor in the short term less valuable because it's likely to be refactored again.